pkg/reconciler/serverlessservice/subsetter.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package serverlessservice
+
+import (
+	"hash"
+	"hash/fnv"
+	"sort"
+	"strconv"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+const (
+	startSalt = "start-angle-salt"
+	stepSalt  = "step-angle-salt"
+	fromSalt  = "from-salt"
+
+	// universe represents the possible range of angles [0, universe).
+	// We want to have universe divide total range evenly to reduce bias.
+	universe = (1 << 10)
+)
+
+// computeAngle returns a uint64 number swhich represents
+// a hashe built off the given `n` string for consistent selection
+// algorithm.
+func computeHash(n string, h hash.Hash64) uint64 {
```

Comment:
Should this return an `int` instead? You're doing that conversion everywhere currently.

Reply:
Thanks for pointing out 👍. There are bugs :)
We can cast to int since on some platforms int can be 32 thus losing precision. Fixed to cast only after taking the modulo.

Reply:
I see, it might make sense to add a comment here stating that so nobody comes through and "cleans" that up.

Reply:
Done.